ArduCopter/mode_guided.cpp

```diff
@@ -584,6 +599,41 @@ void ModeGuided::set_velaccel(const Vector3f& velocity, const Vector3f& accelera
 #endif
 }
 
+// set_actuator_mode - sets guided mode's individual motor mode
+void ModeGuided::set_actuator_mode(float actuator[AP_MOTORS_MAX_NUM_MOTORS]) {
+#if FRAME_CONFIG == MULTICOPTER_FRAME
+    if (guided_mode != SubMode::Actuator) {
+        motor_control_start();
+    }
+
+    guided_motor_state.update_time_ms = millis();
+    memcpy(guided_motor_state.actuator, actuator, AP_MOTORS_MAX_NUM_MOTORS * sizeof(float));
```

Comment:
```suggestion
    memcpy(guided_motor_state.actuator, actuator, ARRAY_SIZE(guided_motor_state.actuator));
```
If you changed to double in the actuators, then this would not require a change.

Reply:
Hi @Ryanf55 ! So ARRAY_SIZE gives number of elements and because of that it will not copy anything after 8th element(32bytes). Either way you suggest using double(float64) instead of float32?

Reply:
I think double precision would be coverkill.

Reply:
Alright Ryan! Then I'll keep it float32 then